### PR TITLE
Docs search

### PR DESCRIPTION
**Pull request by mcnuttandrew**
This PR adds search functionality to the documentation. The basic architecture of this feature works as follows.

1) Before the app arrives in the browser all of the markdown files are compiled into a single index.json file
2) On load of app in the browser a webWorker is launched who grabs the index.json file
3) Individual queries are sent to the web worker for search.

search

This addresses #528

### CODE REVIEW

File: docs/markdown/getting-started/getting-started.md [hunk 1]
```diff
@@ -1,120 +0,0 @@
-### Getting started
```
Comment (mcnuttandrew) [inline docs/markdown/getting-started/getting-started.md]: this file wasn't being used

**Comment by Apercu** [comment]
Ah this is so cool! I'm in the process of creating some sort of project allowing us to bootstrap our documentation website, this will definitely belong in there.

I'll review and accept, dunno how much time it will take to port react-vis to it

**Comment by mcnuttandrew** [comment]
> I'm in the process of creating some sort of project allowing us to bootstrap our documentation website, this will definitely belong in there.

I've been thinking about a project like that! I think a bootstrapping thing could be cool, but I feel like this who structure could be extracted into a gitbook like framework. I imagine such a structure being something analogous to a react component that accepts a JSON list of pages (delimited by section) and style props, and requires the user to keep all of their markdown in a single place. Were i to name to this something, it would be DocsBuilder.

**Comment by Apercu** [comment]
Pardon my wording, it's actually not only a scaffolding tool but also more like a create-react-app sort of thing, abstracting all the configs and base stuff like the redux provider and routes. I'm gonna add you to the repo

**Pull request closed without merging**